tasks/win_agent.yml

```diff
@@ -1,41 +1,39 @@
 ---
-tasks:
-    - name: "Cohesity agent: Set Agent to state of {{ cohesity_agent.state | default('present') }}"
-      cohesity_win_agent:
-        cluster: "{{ cohesity_server }}"
-        username: "{{ cohesity_admin }}"
-        password: "{{ cohesity_password }}"
-        validate_certs: "{{ cohesity_validate_certs | default(False) }}"
-        state: "{{ cohesity_agent.state }}"
-        service_user: "{{ cohesity_agent.service_user | default('') }}"
-        service_password: "{{ cohesity_agent.service_password | default('') }}"
-        preservesettings: "{{ cohesity_agent.preservesettings | default(False) }}"
-        install_type: "{{ cohesity_agent.install_type | default('volcbt') }}"
-      tags: always
-      notify:
-          - Reboot the Hosts after agent modification
+- name: "Cohesity agent: Set Agent to state of {{ cohesity_agent.state | default('present') }}"
+  cohesity_win_agent:
+    cluster: "{{ cohesity_server }}"
+    username: "{{ cohesity_admin }}"
+    password: "{{ cohesity_password }}"
+    validate_certs: "{{ cohesity_validate_certs | default(False) }}"
+    state: "{{ cohesity_agent.state }}"
+    service_user: "{{ cohesity_agent.service_user | default('') }}"
+    service_password: "{{ cohesity_agent.service_password | default('') }}"
+    preservesettings: "{{ cohesity_agent.preservesettings | default(False) }}"
+    install_type: "{{ cohesity_agent.install_type | default('volcbt') }}"
+  tags: always
+  register: installed
 
-    - name: Firewall rule to allow CohesityAgent on TCP port 50051
-      win_firewall_rule:
-        name: Cohesity Agent Ansible
-        description:
-          - Automated Firewall rule created by the Cohesity Ansible integration to allow
-          - for the Cohesity Agent to communicate through the firewall.
-        localport: 50051
-        action: allow
-        direction: in
-        protocol: tcp
-        state: "{{ cohesity_agent.state }}"
-        enabled: yes
-      tags: always
+- name: Firewall rule to allow CohesityAgent on TCP port 50051
+  win_firewall_rule:
+    name: Cohesity Agent Ansible
+    description:
+      - Automated Firewall rule created by the Cohesity Ansible integration to allow
+      - for the Cohesity Agent to communicate through the firewall.
+    localport: 50051
+    action: allow
+    direction: in
+    protocol: tcp
+    state: "{{ cohesity_agent.state }}"
+    enabled: yes
+  tags: always
 
 # => This reboot will only be triggered if both of the following conditions are true:
 # => - The registered variable 'installed' returns True when the changed state is queried.
 # => - The user defined variable 'cohesity_win_agent_reboot' returns as True.
-handlers:
-    - name: Reboot the Hosts after agent modification
-      win_reboot:
-        reboot_timeout: 180
-      when:
-        - cohesity_agent.reboot
-      tags: always
+- name: Reboot the Hosts after agent modification # noqa 503
```

Comment:
Can remove noqa 503

Reply:
using # noqa 503, the task is ignored by ansible lint. Had an issue here, since tasks file is used in the playbooks , it cannot have handlers. And according to lint if we condition on .changed that must be an handler. This checks works when we have such conditions in playbooks but not in task files